### PR TITLE
zls: move to by-name

### LINKED ISSUE #290102: Build failure: zls (aarch64-darwin)
### Steps To Reproduce

1. `nix build nixpkgs#zls`

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/njq0jvd63gcb2lydgnr80ydd009ygwws-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
zig build flags: -Dcpu=baseline -Doptimize=ReleaseSafe
error: unable to create compilation: DarwinSdkNotFound
```

### Additional context

Building on aarch64-darwin fails.

### Notify maintainers
@figsoda @moni-dz 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.0.0, macOS 14.0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.14.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by moni-dz**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
ofborg fails:
```
Running phase: buildPhase
zig build flags: -Dcpu=baseline -Doptimize=ReleaseSafe
error: unable to create compilation: DarwinSdkNotFound
```

**Comment by moni-dz** [comment]
should work now with the new darwin sdk rework

**Comment by 0x5a4** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 290196`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>zls</li>
  </ul>
</details>

File: pkgs/by-name/zl/zls/package.nix [hunk 1]
```diff
@@ -21,9 +22,8 @@ stdenv.mkDerivation (finalAttrs: {
     "-Dversion_data_path=${zig_0_13.src}/doc/langref.html.in"
   ];
 
-  nativeBuildInputs = [
-    zig_0_13.hook
-  ];
+  nativeBuildInputs = [ zig_0_13.hook ];
+  buildInputs = [ apple-sdk_11 ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/zl/zls/package.nix:26]:
```suggestion
  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk_11 ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/zl/zls/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/zl/zls/package.nix:26]: And it looks like we will also have to wait for #353702 to be fixed.

**Comment by emilazy** [comment]
`apple-sdk_11` is already the default on `aarch64-darwin`. Are you sure this does anything to fix the build?

**Comment by moni-dz** [comment]
> `apple-sdk_11` is already the default on `aarch64-darwin`. Are you sure this does anything to fix the build?

seems it builds now without it

**Review by einetuer — APPROVED** [review]
looks good :+1:

**Review by emilazy — APPROVED** [review]

**Pull request merged**